src/autowiring/AutoPacket.cpp

```diff
@@ -96,44 +100,115 @@ void AutoPacket::AddSatCounterUnsafe(SatCounter& satCounter) {
         ss << "Cannot add listener for multi-broadcast type " << autowiring::demangle(pCur->id);
         throw std::runtime_error(ss.str());
       }
-
-      entry.m_subscribers.push_back({
-        pCur->is_shared,
-        pCur->is_multi ?
-        DecorationDisposition::Subscriber::Type::Multi :
-        pCur->is_shared ?
-        DecorationDisposition::Subscriber::Type::Optional :
-        DecorationDisposition::Subscriber::Type::Normal,
-        &satCounter
-      });
-      switch (entry.m_state) {
-      case DispositionState::Complete:
+      if (entry.m_state == DispositionState::Complete) {
         // Either decorations must be present, or the decoration type must be a shared_ptr.
         if (!entry.m_decorations.empty() || pCur->is_shared) {
           satCounter.Decrement();
         }
-        break;
-      default:
-        break;
       }
     }
-    if (pCur->is_output) {
-      if(!entry.m_publishers.empty())
-        for (const auto& subscriber : entry.m_subscribers)
-          for (auto pOther = subscriber.satCounter->GetAutoFilterArguments(); *pOther; pOther++) {
-            if (pOther->id == pCur->id && !pOther->is_multi) {
-              std::stringstream ss;
-              ss << "Added identical data broadcasts of type " << autowiring::demangle(pCur->id) << " with existing subscriber.";
-              throw std::runtime_error(ss.str());
+
+    if (pCur->is_rvalue) {
+      if (entry.m_pModifier) {
+        std::stringstream ss;
+        ss << "Added identical rvalue decorations for type " << autowiring::demangle(pCur->id);
```

Comment:
Actually, this is behavior I want to support.  You should only throw an exception when the altitudes are equal.